packages/medusa/src/services/order.ts

```diff
@@ -726,6 +730,40 @@ class OrderService extends TransactionBaseService {
     })
   }
 
+  protected async createGiftCardsFromLineItem_(order: Order, lineItem: LineItem, manager: EntityManager): Promise<Promise<GiftCard>[]> {
+    const createGiftCardPromises: Promise<GiftCard>[] = []
+
+    // LineItem type doesn't promise either the subtotal or quantity. Adding a check here provides
+    // additional type safety/strictness
+    if (!lineItem.subtotal || !lineItem.quantity) return createGiftCardPromises
+
+    // Subtotal is the pure value of the product/variant excluding tax, discounts, etc.
+    // We divide here by quantity to get the value of the product/variant as a lineItem
+    // contains quantity. The subtotal is multiplicative of pure price per product and quantity
+    const taxExclusivePrice = lineItem.subtotal / lineItem.quantity
+    // The tax_lines contains all the taxes that is applicable on the purchase of the gift card
+    // On utilizing the gift card, the same set of taxRate will apply to gift card
+    // We calculate the summation of all taxes and add that as a snapshot in the giftcard.tax_rate column
+    const giftCardTaxRate = lineItem.tax_lines.reduce(
+      (sum, taxLine) => sum + taxLine.rate, 0
+    )
+
+    for (let qty = 0; qty < lineItem.quantity; qty++) {
+      const createGiftCardPromise = this.giftCardService_.withTransaction(manager).create({
```

Comment:
todo: do not call withTransaction at each loop. instead create a const giftCardServiceTx = this. giftCardService_.withTransction and use it in the loop